src/vmm/src/vmm_config/machine_config.rs

```diff
@@ -65,13 +67,17 @@ pub struct VmConfig {
     /// Enables or disabled SMT.
     #[serde(default, deserialize_with = "deserialize_smt")]
     pub smt: bool,
-    /// A CPU template that it is used to filter the CPU features exposed to the guest.
+    /// A CPU template (hard coded) that it is used to filter the
+    /// CPU features exposed to the guest.
     #[serde(
         default,
         deserialize_with = "deserialize_cpu_template",
         skip_serializing_if = "CpuFeaturesTemplate::is_none"
     )]
     pub cpu_template: CpuFeaturesTemplate,
+    /// CPU template to be defined by the user
+    #[serde(skip_serializing)]
+    pub custom_cpu_template: Option<CustomCpuTemplate>,
```

Comment:
Shouldn't we also always skip `cpu_template` field if all info is stored in the vcpu state?

Reply:
That is up for debate. I think in the long term the hard coded CPU template types will be removed completely.